modules/fedimint-mint-client/src/backup/recovery.rs

```diff
@@ -306,38 +270,20 @@ impl MintRestoreInProgressState {
     /// errors via `sender` itself.
     ///
     /// TODO: could be internal to recovery_loop?
-    fn fetch_block_stream<'a>(
+    async fn await_block<'a>(
         api: DynGlobalApi,
         decoders: ModuleDecoderRegistry,
-        epoch_range: Range<u64>,
```

Comment:
**issue (blocking):** What's the reason to remove it? It was there so we can process the blocks/epochs while this stream is downloading them in parallel (with some buffer) as a stream, making things go faster.

Reply:
The previous setting was to process up to 500 epochs in one stream, which is roughly one block so I removed it to just use blocks directly. Saves on complexity for now.

Reply:
While the code was processing the epoch/block next one could have been downloading already, which is a significant improvement.

Oh well.

Reply:
Should we track this somewhere? I remember recovery performance was a problem in the past.

Reply:
The performance of the new recovery may have changed significantly. We now only download, verify and aggregate the blind signature shares for our unspent nonces; therefore we save a lot of bandwidth and possibly noticeable amount of compute. This might help with previous performance issues already. So a lot of variables are changed now.